### ISSUE TITLE
ERROR getting cluster client with get_secret

### ISSUE THREAD

**Issue by gruyaume** [canonical/vault-k8s-operator#43]
**Describe the bug**
ops throws model errors when running get_secret. This error can be seen intermitently in integration tests. Look [here](https://github.com/canonical/vault-k8s-operator/actions/runs/6302623499/attempts/2) for example.

**To Reproduce**
1. Run integration tests in CI

**Expected behavior**
No error thrown.

**Logs**

```
File "/var/lib/juju/agents/unit-vault-k8s-2/charm/./src/charm.py", line 193, in _on_config_changed
    root_token, unseal_keys = self._get_initialization_secret_from_peer_relation()
  File "/var/lib/juju/agents/unit-vault-k8s-2/charm/./src/charm.py", line 602, in _get_initialization_secret_from_peer_relation
    juju_secret = self.model.get_secret(id=juju_secret_id)
  File "/var/lib/juju/agents/unit-vault-k8s-2/charm/venv/ops/model.py", line 272, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-vault-k8s-2/charm/venv/ops/model.py", line 3148, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-vault-k8s-2/charm/venv/ops/model.py", line 2[79](https://github.com/canonical/vault-k8s-operator/actions/runs/6309422294/job/17129305867?pr=30#step:4:80)4, in _run
    raise ModelError(e.stderr)
ops.model.ModelError: ERROR getting cluster client: unable to determine legacy status for namespace "test-integration-x3js": namespaces "test-integration-x3js" is forbidden: User "system:serviceaccount:test-integration-x3js:unit-vault-k8s-2" cannot get resource "namespaces" in API group "" in the namespace "test-integration-x3js"
```

**Environment**
 - Vault version: 1.13.3
 - Juju version: 3.1.5
 - Cloud Environment: MicroK8s
 - Kubernetes version: 1.27-strict/stable

**Comment by gruyaume** [comment]
This seems to be related to a bug in Juju:
- https://bugs.launchpad.net/juju/+bug/2032686

**Comment by gruyaume** [comment]
This seems to be fixed with Juju 3.1.6
